src/katgpucbf/xbgpu/engine.py

```diff
@@ -164,14 +167,8 @@ def __init__(self, output: Output, engine: "XBEngine", context: AbstractContext,
             f"{output.name}.tx_free_item_queue", self.n_tx_items
         )
 
-    # TODO: Is it fair to impose this on the user?
-    @abstractmethod
-    def _populate_sensors(self) -> None:
-        """Define any Pipeline-specific sensors."""
-        raise NotImplementedError  # pragma: nocover
-
     def add_rx_item(self, item: RxQueueItem) -> None:
-        """Append a newly-received :class:`RxQueueItem`."""
+        """Append a newly-received :class:`RxQueueItem` to the `rx_item_queue`."""
```

Comment:
```suggestion
        """Append a newly-received :class:`RxQueueItem` to the :attr:`_rx_item_queue`."""
```
(haven't actually tested that that renders/links correctly)

Reply:
This one is proving a tough nut to crack. It's syntactically similar to e.g. [this reference in fgpu](https://github.com/ska-sa/katgpucbf/blob/main/src/katgpucbf/fgpu/engine.py#L144), but fails to include the actual/needed hypertext anchor to `#katgpucbf.xbgpu.engine.Pipeline._rx_item_queue`
* Perhaps the `_rx_item_queue` needs to be mentioned as a class attribute like `n_rx/tx_items`?

Perhaps++ this is part of the reasoning behind fgpu's `Pipeline.add_in_item` docstring simply saying
> Append a newly-received :class:`~.InItem`.

* I should also add, the mention of [fgpu's _out_queue](https://github.com/ska-sa/katgpucbf/blob/main/src/katgpucbf/fgpu/engine.py#L926) is only ever done in 'plain old comments' - no rST/sphinx directives.